### PR TITLE
feat(cb2-8823): add way to create provisional on update endpoint

### PR DESCRIPTION
**Pull request by naathanbrown**
## Description

The update end point needs to not always archive, such as when creating a provisional from a current. This adds that functionality. 

Related issue: [8823](https://dvsa.atlassian.net/browse/CB2-8823)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/handler/update.ts [hunk 1]
```diff
@@ -57,9 +56,10 @@ export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayPr
       return addHttpHeaders(vinErrors);
     }
 
+    const archiveNeeded = !((recordFromDB.techRecord_statusCode === StatusCode.CURRENT && requestBody.techRecord_statusCode === StatusCode.PROVISIONAL));
     const [updatedRecordFromDB, updatedNewRecord] = await processUpdateRequest(recordFromDB, requestBody, userDetails);
 
-    const record = await updateVehicle([updatedRecordFromDB] as TechRecordGet[], updatedNewRecord as TechRecordGet);
+    const record = await updateVehicle([updatedRecordFromDB] as TechRecordGet[], updatedNewRecord as TechRecordGet, archiveNeeded);
```
Comment (gjulien-bjss) [inline src/handler/update.ts:62]: can you not just give the records to archive array an empty array if there are no records to archive?

**Review by shivangidas — APPROVED** [review]

**Review by shivangidas — APPROVED** [review]

File: src/handler/update.ts (hunk 1 above)
Comment (naathanbrown) [reply src/handler/update.ts]: sure, check that now

File: src/handler/update.ts [hunk 2]
```diff
@@ -57,9 +56,10 @@ export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayPr
       return addHttpHeaders(vinErrors);
     }
 
+    const archiveNeeded = !((recordFromDB.techRecord_statusCode === StatusCode.CURRENT && requestBody.techRecord_statusCode === StatusCode.PROVISIONAL));
```
Comment (gjulien-bjss) [inline src/handler/update.ts:59]:
```suggestion
    const archiveNeeded = !(recordFromDB.techRecord_statusCode === StatusCode.CURRENT && requestBody.techRecord_statusCode === StatusCode.PROVISIONAL);
```

File: src/handler/update.ts [hunk 3]
```diff
@@ -57,9 +56,10 @@ export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayPr
...
     const [updatedRecordFromDB, updatedNewRecord] = await processUpdateRequest(recordFromDB, requestBody, userDetails);
 
-    const record = await updateVehicle([updatedRecordFromDB] as TechRecordGet[], updatedNewRecord as TechRecordGet);
+    const record = await updateVehicle(archiveNeeded ? [updatedRecordFromDB] as TechRecordGet[] : [], updatedNewRecord as TechRecordGet);
```
Comment (gjulien-bjss) [inline src/handler/update.ts:62]: nitpick:

would be nicer as
```suggestion
    const recordsToArchive = archiveNeeded ? [updatedRecordFromDB] as TechRecordGet[] : []
    const record = await updateVehicle(recordsToArchive, updatedNewRecord as TechRecordGet);
```

**Review by gjulien-bjss — APPROVED** [review]

**Review by shivangidas — APPROVED** [review]

**Pull request merged**